guarddog/analyzer/sourcecode/npm-dll-hijacking.yml

```diff
@@ -0,0 +1,75 @@
+rules:
+  - id: npm-dll-hijacking
+    languages:
+      - javascript
+    message: This package manipulates a trusted application into loading a malicious dll
+    metadata:
+      description: Identifies when a trusted application is manipulated into loading a malicious dll
+    pattern-either:
+
+      # dll side-loading
+      - pattern-either:
+        - patterns:
+          - pattern: "$DLL_LOAD"
+          - metavariable-pattern:
+              metavariable: $DLL_LOAD
+              pattern-either:
+                # load inline windows
+                - pattern-regex: (?i).*?\.exe.*?\.dll
```

Comment:
if I understand properly this is matching any string like `file.exe.dll` or `file.exetest.dll`? Can you explain why?

Reply:
you are right, I will push a fix in the regex to require at least one space in between
However in the practice this is not usual since no false positives were found around this one.

Reply:
Great! Yeah the impact is lower, but definitely good to proactively fix it before it becomes an issue